boa/src/builtins/uri/mod.rs

```diff
@@ -0,0 +1,108 @@
+//! This module implements the global `decodeURI` and encodURI functions.
+
+use std::borrow::Borrow;
+
+use crate::value::RcString;
+use crate::{
+    builtins::BuiltIn, object::FunctionBuilder, property::Attribute, value::Value, BoaProfiler,
+    Context, Result,
+};
+use percent_encoding::{percent_decode, utf8_percent_encode, AsciiSet, CONTROLS};
+
+type EncodeFuncType = fn(&RcString) -> Value;
+
+// https://url.spec.whatwg.org/#fragment-percent-encode-set
+const ENCODE_FRAGMENT: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'<').add(b'>').add(b'`');
+
+fn decode(str: &RcString) -> Value {
+    Value::string(
+        percent_decode(str.as_bytes())
+            .decode_utf8()
+            .unwrap()
+            .borrow(),
+    )
+}
+
+fn encode(str: &RcString) -> Value {
+    Value::string(utf8_percent_encode(str, ENCODE_FRAGMENT).to_string())
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Uri;
+
+impl BuiltIn for Uri {
```

Comment:
I don't think we should implement `BuiltIn` for `Uri`, this means there exists a global `Uri` object, which as far as I know it's not the case, right?

I would just implement an `init()` function, but without implementing the `BuiltIn` trait.

This would allow you to define an `init()` function that doesn't need to return anything in particular.

Reply:
I have removed `BuiltIn` trait but I think we still need to return tuple from `init` because of [this](https://github.com/boa-dev/boa/blob/master/boa/src/builtins/mod.rs#L102).